### PR TITLE
:seedling: E2E tests for clients/githubrepo/languages_e2e_test.go

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Included e2e tests for clients/githubrepo/languages_e2e_test.go

#### What kind of change does this PR introduce?

(Is it a bug fix, feature, docs update, something else?)

- [ ] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note

```

### CODE REVIEW

File: clients/githubrepo/languages_e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		httpClient := &http.Client{
+			Transport: rt,
+		}
+		langHandler = &languagesHandler{
+			ghclient: github.NewClient(httpClient),
+			ctx:      ctx,
+		}
+	})
+	Context("listProgrammingLanguages()", func() {
+		skipIfTokenIsNot(patTokenType, "GITHUB_TOKEN only")
```
Comment (spencerschrock) [inline clients/githubrepo/languages_e2e_test.go:43]: same feedback from earlier PRs about the text mismatching the token type here.

Have we checked that this doesn't work with a certain type of token?

File: go.mod [hunk 2]
```diff
@@ -172,7 +172,7 @@ require (
 	golang.org/x/crypto v0.7.0 // indirect
 	golang.org/x/exp v0.0.0-20230425010034-47ecfdc1ba53
 	golang.org/x/net v0.10.0 // indirect
-	golang.org/x/oauth2 v0.7.0 // indirect
+	golang.org/x/oauth2 v0.7.0
```
Comment (spencerschrock) [inline go.mod:175]: this seems unintentional

**Review by spencerschrock — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (naveensrinivasan) [reply go.mod:175]: This was go mod tidy

File: clients/githubrepo/languages_e2e_test.go (hunk 1 above)
Comment (naveensrinivasan) [reply clients/githubrepo/languages_e2e_test.go]: Thanks, I have now removed check for `skipIfTokenIsNot`

**Pull request merged**